File: frontend/src/api/lms.ts
import { apiClient } from "./client";

export interface ClassroomConnectionResponse {
  id: string;
  provider: string;
  created_at: string;
  expires_at?: string | null;
  profile?: Record<string, string> | null;
}

export interface ClassroomPushResponse {
  id: string;
  status: string;
  external_assignment_id?: string | null;
  created_at: string;
}

export const connectGoogleClassroom = async () => {
  const { data } = await apiClient.post<ClassroomConnectionResponse>(
    "/lms/google-classroom/connect",
    {
      access_token: "stub-access-token",
      refresh_token: "stub-refresh-token",
      expires_in: 3600,
      profile: { teacher: "You" }
    }
  );
  return data;
};

export const pushGoogleClassroomAssignment = async (
  lessonId: string,
  params: { courseId: string; topicId?: string; dueDate?: string }
) => {
  const { data } = await apiClient.post<ClassroomPushResponse>(
    "/lms/google-classroom/push",
    {
      lesson_id: lessonId,
      course_id: params.courseId,
      topic_id: params.topicId,
      due_date: params.dueDate
    }
  );
  return data;
};

File: backend/app/schemas/lms.py
"""Schemas for LMS integration endpoints."""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ClassroomConnectRequest(BaseModel):
    access_token: str
    refresh_token: str | None = None
    expires_in: int | None = Field(default=3600, ge=0)
    profile: dict[str, str] | None = None


class ClassroomConnectResponse(BaseModel):
    id: UUID
    provider: str
    created_at: datetime
    expires_at: datetime | None
    profile: dict[str, str] | None = None

    class Config:
        from_attributes = True


class ClassroomPushRequest(BaseModel):
    lesson_id: UUID
    course_id: str
    topic_id: str | None = None
    due_date: datetime | None = None


class ClassroomPushResponse(BaseModel):
    id: UUID
    status: str
    external_assignment_id: str | None
    created_at: datetime

    class Config:
        from_attributes = True
